backend/layers/persistence/persistence_mock.py

```diff
@@ -267,7 +267,7 @@ def get_collection_version_with_datasets(
         version = self.collections_versions.get(version_id.id)
         if not version:
             return None
-        version = self._update_version_with_canonical(version, update_datasets=True)
+        version = self._update_version_with_canonical(version, update_datasets=True, get_tombstoned=get_tombstoned)
```

Comment:
this is a great catch. We should introduce this fix independent of any other concerns.